src/main/java/com/everis/control/CarService.java

```diff
@@ -0,0 +1,66 @@
+package com.everis.control;
+
+
+import java.util.List;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import com.everis.control.CarService;
+import com.everis.entity.Car;
+
+
+public class CarService {
+
+    @PersistenceContext(unitName = "car-unit")
+    private EntityManager em;
+
+    /**
+     * Method to get a list of Car Entity available
+     */
+    public List<Car> getCars()
+    {	
+		List<Car> listCars = em.createNamedQuery("Car.findAll", Car.class).getResultList();
+		return listCars;
+    }
+
+    /**
+     * Method to get one Car info using its id                             
+     */
+    public Car getCar(int id)
+    { 	
+    	Car car = em.createNamedQuery("Car.findById", Car.class).setParameter("id", id).getSingleResult();
+    	if (car != null) { 					
+    	    return car;
+
+    	} else { 	    
+    	    return new Car();
```

Comment:
This is a bit strange, if the car is not found then it returns a Car object ?
Usually an exception is thrown or a null value is returned.